include/engine/datafacade/contiguous_internalmem_datafacade.hpp

```diff
@@ -382,20 +382,20 @@ class ContiguousInternalMemoryDataFacadeBase : public BaseDataFacade
     std::vector<PhantomNodeWithDistance>
     NearestPhantomNodes(const util::Coordinate input_coordinate,
                         const size_t max_results,
-                        const boost::optional<double> max_distance,
-                        const boost::optional<Bearing> bearing,
+                        const std::optional<double> max_distance,
```

Comment:
Passing an optional is arguably cheap, but would it make sense to just pass a const ref?

Reply:
Well, tbh I believe for primitive like double it won't make any difference(or even make things a bit worse due to indirection)... WDYT?